src/main/java/com/google/sps/model/GmailClient.java

```diff
@@ -42,6 +45,18 @@
    */
   Message getUserMessage(String messageId, MessageFormat format) throws IOException;
 
+  /**
+   * Gets a message from a user's Gmail account. Format assumed to be METADATA. Returned message
+   * will only include email message ID, labels, and specified headers
+   *
+   * @param messageId messageID (retrieved from listUserMessages) of the desired Message
+   * @param metadataHeaders list of names of headers (e.g. "From") that should be included
+   * @return a Message object with the requested information
+   * @throws IOException if an issue occurs with the Gmail service
+   */
+  Message getUserMessageWithMetadataHeaders(String messageId, List<String> metadataHeaders)
```

Comment:
This seems like an implementation details of `GmailClientImpl`. Does it need to be exposed via the interface?

Reply:
This actually is a supported get operation from the Gmail API (where metadataHeaders can simply be passed as a parameter), so I think this belongs here semantically.

I'm hoping that others also use this, since it is more efficient than the standard METADATA format if you don't need all the headers.